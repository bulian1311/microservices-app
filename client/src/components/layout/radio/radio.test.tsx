import * as React from "react";
import { render, RenderResult } from "@testing-library/react";
import Radio from "./radio.component";

let documentBody: RenderResult;

beforeEach(() => {
  documentBody = render(
    <Radio
      value="test"
      label="test label"
      checked
      handleSelectChange={() => {}}
    />
  );
});

test("Radio matces snapshot.", () => {
  const { baseElement } = documentBody;
  expect(baseElement).toMatchSnapshot();
});

test("Show label.", () => {
  expect(documentBody.getByText("test label")).toBeInTheDocument();
});

test("Radio checked.", () => {
  const { getByTestId } = render(
    <form>
      <Radio
        testId="test1"
        value="test1"
        label="label1"
        checked={true}
        handleSelectChange={() => {}}
      />
      <Radio
        testId="test2"
        value="test2"
        label="label2"
        checked={false}
        handleSelectChange={() => {}}
      />
    </form>
  );

  const radio1 = getByTestId("test1") as HTMLInputElement;
  const radio2 = getByTestId("test2") as HTMLInputElement;

  expect(radio1).toBeChecked();
  expect(radio2).not.toBeChecked();
});
