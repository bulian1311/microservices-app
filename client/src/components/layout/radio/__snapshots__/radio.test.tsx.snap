// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio matces snapshot. 1`] = `
<body>
  <div>
    <div
      class="sc-bdfBwQ vQKso"
    >
      <input
        checked=""
        class="sc-dlfnbm dmZjux"
        type="radio"
        value="test"
      />
      <label
        class="sc-gsTCUz kpecLn"
      />
      <span
        class="sc-hKgILt jeJdIm"
      >
        test label
      </span>
    </div>
  </div>
</body>
`;
